Platformio/HAL/HardwareModules/wifiHandlerInterface.h

```diff
@@ -1,14 +1,24 @@
 #pragma once
 #include <string>
-#include "HardwareAbstract.hpp"
+#include <memory>
+#include <functional>
+
+typedef struct {
+  std::string ssid;
+  int rssi;
+} WifiInfo;
+
+typedef struct {
+  bool isConnected;
+  std::string IP;
+  std::string ssid;
+}wifiStatus;
 
 class wifiHandlerInterface{
     public:
-        virtual void begin() = 0;
-        //virtual void connect(const char* SSID, const char* password) = 0;
-        virtual void disconnect() = 0;
-        virtual bool isConnected() = 0;
-        virtual void turnOff() = 0;
+        virtual bool isAvailable() = 0;
         virtual void scan() = 0;
-        virtual std::string getIP() = 0;
+        virtual void connect(std::shared_ptr<std::string> ssid, std::shared_ptr<std::string> password) = 0;
```

Comment:
you have a specific reason for shared_ptr here? think it should be okay without.

Reply:
so the string is not actually copied. If we have just a std::string, the compiler would copy the string which means the memory usage would depend on the string size instead of copying the shared pointer which is always the same. But we could also use std::string here if this would be better

Reply:
Yeah I guess I am under the assumption that the strings here would be pretty small typically.